### PR TITLE
H5md proposol conform

### PR DESCRIPTION
**Pull request by KaiSzuttor**
PR Checklist
------------
 - [x] Tests?
   - [x] Interface

### CODE REVIEW

File: src/core/io/writer/h5md/h5md_core.hpp [hunk 1]
```diff
@@ -39,180 +38,187 @@
...
+  std::unordered_map<std::string, h5xx::dataset> datasets;
 };
 
-
 struct incompatible_h5mdfile : public std::exception {
-    const char* what () const throw ()
-    {
-        return "The given hdf5 file does not have a valid h5md structure!";
-    }
+  const char *what() const throw() {
```
Comment (KaiSzuttor) [inline src/core/io/writer/h5md/h5md_core.hpp]: Isn't "throw()" already telling the compiler that this function does not throw anything, therefore "noexcept" is not needed here?

File: src/core/io/writer/h5md/h5md_core.hpp [hunk 2]
```diff
@@ -39,180 +38,187 @@
...
+   */
+  void Write(int write_dat);
+
+  /**
+   * @brief Method to write the energy contributions to the H5MD file.
+   * @param Boolean values for total, kinetic.
+   * \todo Implement this method.
+   */
+  void WriteEnergy(bool total = true, bool kinetic = true);
+  std::string &filename() { return m_filename; };
```
Comment (KaiSzuttor) [inline src/core/io/writer/h5md/h5md_core.hpp]: this is due to the scriptinterface access, I just adopted this structure from e.g. shapes/Wall.hpp.

File: src/core/io/writer/h5md/h5md_core.hpp [hunk 3]
```diff
@@ -39,180 +38,187 @@
...
+  /**
+   * @brief Method to force flush to h5md file.
+   */
+  void Flush();
+
+private:
+  MPI_Comm m_hdf5_comm;
+  bool m_already_wrote_bonds = false;
+
+  bool check_file_exists(const std::string &name) {
```
Comment (hmenke) [inline src/core/io/writer/h5md/h5md_core.hpp]: This function is unused now.  Perhaps remove?

File: src/core/io/writer/h5md/h5md_core.hpp (hunk 2 above)
Comment (fweik) [inline src/core/io/writer/h5md/h5md_core.hpp]: This is technically no necessary, you could e.g. return a const& and make the property read only (which a think it is?)

**Pull request merged**